hub/core/dataset/dataset.py

```diff
@@ -1361,9 +1362,17 @@ def pytorch(
 
         Returns:
             A torch.utils.data.DataLoader object.
+
+        Raises:
+            EmptyTensorInTheDatasetError - when one of the tensors is empty
         """
         from hub.integrations import dataset_to_pytorch as to_pytorch
 
+        # check whether we have an empty tensor inside of tensors
+        for tensor_name in self.tensors.keys():
```

Comment:
if tensors are passed to .pytorch then use those, instead of self.tensors.keys.
You can do this check inside dataset_to_pytorch method after this line https://github.com/activeloopai/Hub/blob/main/hub/integrations/pytorch/pytorch.py#L127
instead of doing it in hub/core/dataset/dataset.py